### ISSUE TITLE
A11y - Incorrect heading level in CaptionLabel

### ISSUE THREAD

**Issue by bradleysmagacz** [gpbl/react-day-picker#1607]
### Bug description

Inside of this file: https://github.com/gpbl/react-day-picker/blob/a8c3e3c1565ff2fc729825b2f337a6323bbaae2b/packages/react-day-picker/src/components/CaptionLabel/CaptionLabel.tsx

The tag is hardcoded as h2, but there are times when this component is being used as a lower level heading. In order to fulfill WCAG heading requirements, there should be a way to override the heading level if need be.

### Expected behavior

Allow the consumer of the component to set the proper heading level.
